Arrowgene.Ddon.GameServer/Characters/ItemManager.cs

```diff
@@ -727,6 +734,33 @@ public static bool SendToItemBag(uint storageType)
             }
             return toBag;
         }
+
+        public List<CDataItemUpdateResult> RemoveAllItemsFromInventory(Character character, Storages storages, List<StorageType> storageTypes, DbConnection connection = null)
+        {
+            var results = new List<CDataItemUpdateResult>();
+            foreach (var storageType in storageTypes)
+            {
+                for (int i = 0; i < character.Storage.GetStorage(storageType).Items.Count; i++)
+                {
+                    ushort slotNo = (ushort)(i + 1);
+
+                    var storageItem = storages.GetStorage(storageType).GetItem(slotNo);
+                    if (storageItem != null)
+                    {
+                        results.Add(_Server.ItemManager.CreateItemUpdateResult(null, storageItem.Item1, storageType, slotNo, 0, 0));
+                        results.Add(_Server.ItemManager.CreateItemUpdateResult(null, storageItem.Item1, storageType, slotNo, 0, storageItem.Item2));
+                    }
+
+                    character.Storage.GetStorage(storageType).SetItem(null, 0, slotNo);
+                    if (connection != null)
+                    {
+                        _Server.Database.DeleteStorageItem(character.ContentCharacterId, storageType, slotNo, connection);
```

Comment:
So the original function didn't save it in the DB because there was another query which used the transaction to delete everything in one fell swoop. I suppose I can change that though.